frontend_tests/node_tests/composebox_typeahead.js

```diff
@@ -1458,18 +1458,26 @@ run_test('typeahead_results', () => {
     // Autocomplete user mentions by user name.
     assert_mentions_matches('cordelia', [cordelia]);
     assert_mentions_matches('cordelia le', [cordelia]);
+    assert_mentions_matches('co le', [cordelia]);
     assert_mentions_matches('cordelia le ', []);
     assert_mentions_matches('King ', [hamlet, lear]);
     assert_mentions_matches('King H', [hamlet]);
     assert_mentions_matches('King L', [lear]);
     assert_mentions_matches('delia lear', []);
     assert_mentions_matches('Mark Tw', [twin1, twin2]);
+    assert_mentions_matches('o t m o v', [othello]);
+    assert_mentions_matches('o the moor', [othello]);
+    assert_mentions_matches('o te m', []);
+    assert_mentions_matches('t the moor', []);
+    assert_mentions_matches('the Moor of Venice', []);
```

Comment:
This query should return the Othello as well. Say, in a real world scenario if I search for `oth venice` I would expect Othello as a result.

Essentially, I think we should match the name "Othello the Moor of Venice" with code equivalent to the regex: `/oth*\svenice*/ instead of matching each search term with its respective index's word in the name only.